### PR TITLE
Cody PLG: Add Gemini 1.5 Pro and Gemini 1.5 Flash models

### PR DESCRIPTION
**Pull request by abeatrix**
CLOSE https://github.com/sourcegraph/cody-issues/issues/212
CLOSE https://github.com/sourcegraph/cody-issues/issues/411

Server-side change completed in https://github.com/sourcegraph/sourcegraph/pull/62927 & https://github.com/sourcegraph/sourcegraph/pull/63053

This change adds two new AI models to the DEFAULT_DOT_COM_MODELS array:
- Gemini 1.5 Pro
- Gemini 1.5 Flash

These models are provided by Google and are available to Cody Pro users only.

The changes also update the current chat clients to handle Gemini models added by users through the `cody.dev.models` and the ones provided by Cody Gateway.

## Test plan

1. Start Cody from this branch in debug mode and sign in as a Cody Pro users
2. Start a new chat to verify `Gemini 1.5 Pro and Gemini 1.5 Flash` are in the model dropdown list
3. Send a question to both models to confirm they are working correctly:
  - Gemini 1.5 Pro
  - Gemini 1.5 Flash

image

4. Verify the new models are also showing up in the Edit models dropdown list:

image

5. If you have Gemini models added with `cody.dev.models` using your own API key, you should also see the models listed in the dropdown list under the `Other` option:

image

### CODE REVIEW

**Comment by PriNova** [comment]
Any plans to include Codestral as FIM and instruct code model?

**Comment by abeatrix** [comment]
@PriNova I don't have any info on that unfortunately, but feel free to file a feature request on our community site if this is something you want 😄

File: lib/shared/src/llm-providers/clients.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { ModelUIGroup, googleChatClient, groqChatClient, ollamaChatClient } from '..'
+import { type Model, ModelsService } from '../models'
+import type { CompletionLogger } from '../sourcegraph-api/completions/client'
+import type { CompletionCallbacks, CompletionParameters } from '../sourcegraph-api/completions/types'
+
+export async function useCustomChatClient(
```
Comment (abeatrix) [inline lib/shared/src/llm-providers/clients.ts:6]: this is just exisiting code moved from vscode/src/completions/nodeClient.ts

**Review by slimsag — APPROVED** [review]

**Comment by PriNova** [comment]
Nice. Thank you

**Pull request merged**